### PR TITLE
Upgrade operator-sdk to 1.4.2

### PR DESCRIPTION
**Pull request by andrewazores**
https://v1-5-x.sdk.operatorframework.io/docs/upgrading-sdk-version/v1.4.0/

Pretty straightforward upgrade this time, thankfully. I did a bit of manual testing and changing the finalizer names doesn't seem to have broken them, and everything else appears to still be working.

### CODE REVIEW

File: bundle/manifests/container-jfr-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -143,7 +143,7 @@ spec:
                   valueFrom:
                     fieldRef:
                       fieldPath: metadata.annotations['olm.targetNamespaces']
-                image: quay.io/rh-jmc-team/container-jfr-operator:1.0.0-beta5
+                image: quay.io/andrewazores/container-jfr-operator:1.0.0-beta5
```
Comment (ebaron) [inline bundle/manifests/container-jfr-operator.clusterserviceversion.yaml:146]: This must have slipped in.

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -12,5 +12,5 @@ apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
 images:
 - name: controller
-  newName: quay.io/rh-jmc-team/container-jfr-operator
+  newName: quay.io/andrewazores/container-jfr-operator
```
Comment (ebaron) [inline config/manager/kustomization.yaml:15]: This too.

**Comment by andrewazores** [comment]
Should I go ahead and start on 1.5.0 as well?

https://github.com/rh-jmc-team/container-jfr-operator/pull/176#issuecomment-801081815

**Review by ebaron — APPROVED** [review]

**Comment by ebaron** [comment]
> Should I go ahead and start on 1.5.0 as well?
> 
> [#176 (comment)](https://github.com/rh-jmc-team/container-jfr-operator/pull/176#issuecomment-801081815)

Sure, it seems like it will be useful for us.

**Pull request merged**